back-end/aws-cloud9-api/src/main/java/com/serverless/dal/QuestionDBTable.java

```diff
@@ -0,0 +1,199 @@
+public class QuestionDBTable {
+
+private static final String QUESTIONS_TABLE_NAME = System.getenv("QUESTIONS_TABLE_NAME");
+    
+	private static DynamoDBAdapter db_adapter;
+    private final AmazonDynamoDB client;
+    private final DynamoDBMapper mapper;
+
+
+
+    
+
+    
+    
+   
+
+	public QuestionDBTable() {
+	        // build the mapper config
+	        DynamoDBMapperConfig mapperConfig = DynamoDBMapperConfig.builder()
+	            .withTableNameOverride(new DynamoDBMapperConfig.TableNameOverride(QUESTIONS_TABLE_NAME))
+	            .build();
+	        // get the db adapter
+	        this.db_adapter = DynamoDBAdapter.getInstance();
+	        this.client = this.db_adapter.getDbClient();
+	        // create the mapper with config
+	        this.mapper = this.db_adapter.createDbMapper(mapperConfig);
+	}
+	
+	   public List<Question> list() throws IOException {
+	    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
+	        List<Question> results = this.mapper.scan(Question.class, scanExp);
+	        return results;
+	    }
+	   
+	   
+	   public List<ClosedQuestion> listclosed() throws IOException {
+	    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
+	        List<ClosedQuestion> results = this.mapper.scan(ClosedQuestion.class, scanExp);
+	        return results;
+	    }
+
+	
+	    
+	
+
+	
+	// methods
+    public Boolean ifTableExists() {
+        return this.client.describeTable(QUESTIONS_TABLE_NAME).getTable().getTableStatus().equals("ACTIVE");
+    }
+    
+ 
+    public ClosedQuestion get(String id) throws IOException {
+    	ClosedQuestion question = null;
+
+        HashMap<String, AttributeValue> av = new HashMap<String, AttributeValue>();
+        av.put(":v1", new AttributeValue().withS(id));
+
+        DynamoDBQueryExpression<ClosedQuestion> queryExp = new DynamoDBQueryExpression<ClosedQuestion>()
+            .withKeyConditionExpression("id = :v1")
+            .withExpressionAttributeValues(av);
+
+        PaginatedQueryList<ClosedQuestion> result = this.mapper.query(ClosedQuestion.class, queryExp);
+        if(result.size() > 0) {
+        	question = result.get(0);
+        }
+        return question;
+    }
+    
+  
+    
+    public  List<ClosedQuestion> getFormQuestions(String formid) throws IOException {
+    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
+        List<ClosedQuestion> results = this.mapper.scan(ClosedQuestion.class, scanExp);
+        List<ClosedQuestion> res = new ArrayList<ClosedQuestion>();
+        for (int i=0; i< results.size(); i++)
+        {
+        	if (results.get(i).getForm_membership()!=null) {
+        	if(results.get(i).getForm_membership().equals(formid))
+        	{
+        		res.add(results.get(i));
+        	}
+        	}
+        }
+        
+        
+        return res;
+    }
+    
+    public  Boolean deleteQuestionFromForm(String questionid) throws IOException {
```

Comment:
Myślę, że lepiej by było, gdyby funkcja zwracała obiekt, który usuwa, potem można ten obiekt ustawić w body odpowiedzi, żeby się wyświetlił w konsoli

Reply:
Chociaż w sumie nie wiem czy zmiana jest potrzebna. Może zostawić to tak jak jest?